src/Core/Runner.cpp

```diff
@@ -143,14 +126,36 @@ void Runner::runDetached(const QString &tmpFilePath, const QString &sourceFilePa
     LOG_INFO("Running apple script\n" << script);
     runProcess->write(script.toUtf8());
     runProcess->closeWriteChannel();
-#else
+#elif defined(Q_OS_WIN)
     // use cmd on Windows
     runProcess->start("cmd", QProcess::splitCommand(
                                  "/C \"start cmd /C " +
                                  getCommand(tmpFilePath, sourceFilePath, lang, runCommand, args).replace("\"", "^\"") +
                                  " ^& pause\""));
     LOG_INFO("CMD Arguemnts " << runProcess->arguments().join(" "));
+#else
+    QString terminal = SettingsHelper::getTerminalProgram();
 
+    LOG_INFO("Using: " << terminal << " on Linux");
+    QProcess testProcess;
+    testProcess.start(terminal, {"-v"});
+    bool finished = testProcess.waitForFinished(2000);
+    if (!finished || testProcess.exitCode() != 0)
```

Comment:
Not all terminal emulators support `-v`, nor all terminal emulators support `--version`

It could be better if we handle the `errorOccurred` signal by checking whether the error is `QProcess::FailedToStart`. It is useful for both detached execution and normal execution.

Reply:
This is out of the scope of this PR. We can open a new issue for this code refactoring. I suggest we can have a Util function like `isExecutable(QString path)`. The path is either name of program in which case it looks at environment or an absolute Path.

Reply:
> This is out of the scope of this PR.

It was fine with `xterm`, but now it breaks with some terminals. Are you sure we should fix this in another PR?

Reply:
BTW, (though this is out of the scope of this PR), I don't like the diagnostics for compiler and runner. Do you think it's very useful?

Reply:
What I want to change to is not a new way to check whether a program is available, but an error message when we fail to start a program.